addons/mail/static/tests/tours/discuss_public_tour.js

```diff
@@ -66,6 +69,9 @@ registry.category("web_tour.tours").add("mail/static/tests/tours/discuss_public_
         {
             content: "Check message is shown",
             trigger: '.o-mail-message-body:contains("cheese")',
+            async run() {
+                await waitUntil('.o-mail-message-body:contains("cheese")');
+            },
```

Comment:
I also fail to understand how this would fix the race condition.
From my understanding, `trigger: <selector>` already does something equivalent to `waitUntil`. If there's a race condition, then this would just put a micro-task tick awaiting, which in theory slightly minimizes the race condition but it's still present.